reconcile/utils/mr/ocm_upgrade_scheduler_org_updates.py

```diff
@@ -0,0 +1,77 @@
+from ruamel import yaml
+
+from reconcile.utils.mr.base import MergeRequestBase
+from reconcile.utils.mr.labels import AUTO_MERGE
+
+
+class CreateOCMUpgradeSchedulerOrgUpdates(MergeRequestBase):
+
+    name = "create_ocm_upgrade_scheduler_org_updates_mr"
+
+    def __init__(self, updates_info):
+        self.updates_info = updates_info
+
+        super().__init__()
+
+        self.labels = [AUTO_MERGE]
+
+    @property
+    def title(self):
+        return f"[{self.name}] ocm upgrade scheduler org updates"
+
+    def process(self, gitlab_cli):
+        changes = False
+        ocm_path = self.updates_info["path"]
+        ocm_name = self.updates_info["name"]
+
+        raw_file = gitlab_cli.project.files.get(
+            file_path=ocm_path, ref=self.main_branch
+        )
+        content = yaml.load(raw_file.decode(), Loader=yaml.RoundTripLoader)
+        upgrade_policy_clusters = content["upgradePolicyClusters"]
+
+        for update in self.updates_info["updates"]:
+            action = update["action"]
+            cluster_name = update["cluster"]
+            upgrade_policy = update.get("policy")
+
+            if action == "add":
+                found = [
+                    c for c in upgrade_policy_clusters if c["name"] == cluster_name
+                ]
+                if found:
+                    continue
+                item = {
+                    "name": cluster_name,
+                    "upgradePolicy": upgrade_policy,
+                }
+                upgrade_policy_clusters.append(item)
+                changes = True
+            elif action == "delete":
+                found = [
+                    c for c in upgrade_policy_clusters if c["name"] == cluster_name
+                ]
+                if not found:
+                    continue
+                upgrade_policy_clusters = [
+                    c for c in upgrade_policy_clusters if c["name"] != cluster_name
+                ]
+                changes = True
+            else:
+                raise NotImplementedError(action)
+
+        yaml.explicit_start = True  # type: ignore[attr-defined]
+        new_content = yaml.dump(
+            content, Dumper=yaml.RoundTripDumper, explicit_start=True
+        )
+
+        msg = f"update {ocm_name} upgrade policy clusters"
+        gitlab_cli.update_file(
+            branch_name=self.branch,
+            file_path=ocm_path,
+            commit_message=msg,
+            content=new_content,
+        )
+
+        if not changes:
```

Comment:
shouldn't this be moved before the `yaml.dump` and `update_file` calls, with an early `return` ?

Reply:
right. i was copying the example of `clusters_updates`, where we add a commit per cluster update in a single MR (may include multiple updates). in this scenario, we are adding a commit per ocm org and submitting the MR per org, so an early cancel would make more sense.

https://github.com/app-sre/qontract-reconcile/pull/2925/commits/d3e4a202199c5cdbe6707a1e6393772d655264e3

Reply:
I think we're missing a `return` statement now also. (or i am not refreshing fast enough...)

Reply:
Ah I did not see `self.cancel` actually raises an exception